### PR TITLE
fix: resolve accented filesystem path correctly in $RefParser lib

### PR DESCRIPTION
**Pull request by paulRbr**
The internal `$refs` Class of the json-schema-ref-parser library stores
paths in a URI encoded fashion. This is to prevent confusion between #
hash internal references (mentioned in the full json path) and #
characters in the filesystem path.

This commit makes sure to fetch the parsed values from the `$refs`
object from the `$refs.values()` function (instead of the
`$refs.get($ref)` function).

I got a bit confused with the internals of the lib as I expected to be
able to provide the returned values of `$Refs.paths()` to the
`$Refs.get()` function transparently. But this does not work for
filesystem paths containing special characters. The best solution is
thus to fetch the `$refs.values()` object which has decoded file path
as keys. I.e. exactly what we need.

### CODE REVIEW

**Comment by scharrier** [comment]
Could you provide more details about what's happening (from our CLI point of view)? I'm not sure to fully understand the issue.

I don't have any strong opinion on how we should fix this. Relying on a lib that isn't released is generally something we try to avoid, at least until we're sure that the fix is going to be merged. As, for now, we don't have any feedback from the APIDevTools team, I think we should at least wait for this feedback (if the bug isn't too important). WDYT?

**Comment by paulRbr** [comment]
> Could you provide more details about what's happening (from our CLI point of view)? I'm not sure to fully understand the issue.

The CLI currently can't use push any definition file if the filesystem path contains non URI characters (I.e. non-alpha or non-digit as per https://datatracker.ietf.org/doc/html/rfc3986#section-2.3).

Thus, running `bump preview ~/Téléchargements/openapi.yml` will fail with the error:
```
 ›   Error: Error resolving $ref pointer                                                                 
 ›   "/home/user/Téléchargements/openapi.yml".                       
 ›   "/home/user/Téléchargements/openapi.yml" not found.   
```

**Comment by paulRbr** [comment]
@scharrier I've managed to fix without any patch to the upstream library. It was me being wrong and reading the documentation badly 😓.

This patch is now working for the accented filesystem paths.

**Review by thimy — COMMENTED** [review]
LGTM but maybe you could wait for another review :D

File: src/definition.ts
Comment (scharrier) [inline src/definition.ts:141]: Why can we remove this? We are now sure that the main definition can be returned, even if parsing doesn't work?

**Review by scharrier — APPROVED** [review]
This is far more clear, congrats for better understanding the internals of `$RefParser`. LGTM, I left a small question.

File: src/definition.ts
Comment (paulRbr) [reply src/definition.ts:141]: Because we now initialize the variable with an “empty” state: 
`let mainReference: JSONSchemaWithRaw = { parsed: {}, raw: '' };`

So the case where we can't find the main reference will be handled by the error of line 144:

```typescript
    if (!parsed || !(parsed instanceof Object) || !('info' in parsed)) {
      throw new UnsupportedFormat(
        "Definition needs to be an object with at least an 'info' key",
      );
    }
```

**Pull request merged**
